### ISSUE TITLE
Get the possibility to have a sub-list

### ISSUE THREAD

**Issue by zak39** [nextcloud-libraries/nextcloud-vue#4755]
Hi everyone :wave: 

As a web developer, I would like to add a sub-list to a [NcAppNavigationItem](https://nextcloud-vue-components.netlify.app/#/Components/App%20containers/NcAppNavigation?id=ncappnavigationitem) component.

In fact, I would like to get an equivalent of this HTML code : 

```html
<ul>
  <li>Countries</li>
  <ul>
    <li>France</li>
    <ul>
      <li>Paris</li>
      <li>Roubaix</li>
    </ul>
    <li>Geremany</li>
    <ul>
      <li>Berlin</li>
      <li>Munich</li>
    </ul>
  </ul>
</ul>
```

Currently, it's not possible. Because, the [NcAppNavigation](https://github.com/nextcloud-libraries/nextcloud-vue/blob/23007bec81f795467b146f5e2376b137d6da467e/src/components/NcAppNavigation/NcAppNavigation.vue#L67-L69) component gives developers just one level of a `<ul>` html element.

We can write an equivalent to this code : 

```html
<NcAppNavigation>
 <template #list>
   <NcAppNavigationItem name="Countries" />
   <ul>
      <NcAppNavigationItem name="Paris" />
      <NcAppNavigationItem name="Roubaix" />
   </ul>
 </template>
</NcAppNavigation>
```

However, this requires to apply a custom css and it may break Nextcloud's flow design.

What do you think ?

**Comment by susnux** [comment]
This is already possible as `NcAppNavigationItem` allows children, see example:

https://nextcloud-vue-components.netlify.app/#/Components/App%20containers/NcAppNavigation?id=ncappnavigationitem


"Element with children"

**Comment by susnux** [comment]
Do you have a use case for more than one level nesting?
I am curious because nesting more than one level in the app navigation seems to me not a good UX.

**Comment by zak39** [comment]
> Do you have a use case for more than one level nesting? I am curious because nesting more than one level in the app navigation seems to me not a good UX.

Yes of course !

I have this mockup made by a UX colleague : 

![image](https://github.com/nextcloud-libraries/nextcloud-vue/assets/28636549/3bdff5fb-a397-4474-a1f6-46389c10e155)

The idea is that "Project cloud", "Workspace Groups" ("Groupes Workspace" in french) and "Connected Groups" ("Groupes Connectés" in french) are `NcAppNavigationItem`.

However, "Workspace groups" and "Connected Groups" could be `NcAppNavigationCaption` (?)

**Comment by susnux** [comment]
> It depends on the case. In the left sidebar, I agree with you on this point.

Yes but that is what `NcAppNavigation*` is about

> But, if it's in the main content or in a modal window, it could make sense.

I do not think you should use `NcAppNaviation*` in such cases but `NcListItem` or similiar?

---

cc @jancborchardt as this probably is a design decision.

**Comment by jancborchardt** [comment]
We habe this one-subfolder level in Files for "Shares", is that not a component?

**Comment by susnux** [comment]
@jancborchardt yes but this is about arbitrary levels, e.g. "Shares -> Shared by you -> shared to XY -> list of shares".
